src/proto/linear_map.rs

```diff
@@ -0,0 +1,559 @@
+//! A module providing a map implementation `LinearMap` backed by a vector.
+
+#![warn(missing_docs)]
+
+use std::iter::Map;
+use std::iter::RandomAccessIterator;
+use std::mem;
+use std::slice;
+
+// TODO: Unzip the vectors?
```

Comment:
Cache; You spend all your time looking at the keys, then maybe touch a single value.